### PR TITLE
PLT-8634 test structure marconi-sidechain-experimental

### PR DESCRIPTION
**Pull request by brendanrbrown**
Creates a base on which we can add tests of handlers to `marconi-sidechain-experimental`. The general approach is to generate events for a Mockchain and to test JSON RPC query handlers directly, without needing to run a node or the HTTP server.

* Extracts various generators and utilities for creating events from `marconi-cardano-indexers-test`, making them available in `marconi-cardano-indexers-test-lib`.
* Adds generators for the events needed to test handlers to the new modules `marconi-cardano-indexers-test-lib`.
   * These mainly are transformations to use the existing Mockchain generators and create the types needed for indexing events with the types needed for queries tested here.
* Adds a derived `Show` instance for `WithDistance` so these events can be generated with hedgehog.
* Creates module `Test.Gen.Marconi.Cardano.Indexers` with a version of `buildIndexer` that exposes the underlying indexers in the coordinator used for `SidechainHttpConfig`.
   * Needed (for now) so that generated events can be indexed directly without needing to pass through the chain-sync protocol, whose 'BlockEvent' type we cannot generate.
   * The module also includes utilities for directly indexing events to the indexers used for tests in this PR.
* Provides tests of basic query results for the `getUtxosFromAddress` and `getBurnTokenEvents`.
* Sets `CARDANO_NODE_CONFIG` to make the cardano node `config` directory available both in CI checks and the nix shell.

Note:

* Utilities for directly indexing events only cover the needs of the handler tests added here. Future PRs can just extend the existing utilities to support indexing other event types as needed for testing additional handlers.

Pre-submit checklist:
- Branch
    - [X] Tests are provided (if possible)
    - [X] Commit sequence broadly makes sense and have useful messages
    - [ ] Important changes are reflected in changelog.d of the affected packages
    - [X] Relevant tickets are mentioned in commit messages
- PR
    - [ ] (For external contributions) Corresponding issue exists and is linked in the description
    - [X] Targeting main unless this is a cherry-pick backport
    - [X] Self-reviewed the diff
    - [X] Useful pull request description
    - [ ] If relevant, reference the ADR in the PR and reference the PR in the ADR
    - [x] Reviewer requested

### CODE REVIEW

File: marconi-cardano-indexers/test-lib/Test/Gen/Marconi/Cardano/Indexers/UtxoQuery.hs [hunk 1]
```diff
@@ -0,0 +1,136 @@
+{-# LANGUAGE FlexibleContexts #-}
```
Comment (brendanrbrown) [inline marconi-cardano-indexers/test-lib/Test/Gen/Marconi/Cardano/Indexers/UtxoQuery.hs:1]: Noting that this module is entirely extracted from `Test.Spec.Marconi.Cardano.Indexers.UtxoQuery` . Unlike with the others, I did not add anything here.

**Comment by brendanrbrown** [comment]
Note the failure in the final commit appears a fluke: The previous commit's check passed, and the latest commit just changed a code comment. I've requested a rerun so we'll see.

File: marconi-sidechain-experimental/test/Spec/Marconi/Sidechain/Experimental/Utils.hs [hunk 2]
```diff
@@ -1,23 +1,217 @@
...
+    mintTokenToUniform
+      :: MintTokenEvent.MintTokenEvent
+      -> (C.BlockNo, C.TxId, C.AssetName, C.Quantity, Maybe MintTokenEvent.MintAssetRedeemer)
+    mintTokenToUniform e =
+      ( e ^. MintTokenEvent.mintTokenEventLocation . MintTokenEvent.mintTokenEventBlockNo
+      , e ^. MintTokenEvent.mintTokenEventLocation . MintTokenEvent.mintTokenEventTxId
+      , e ^. MintTokenEvent.mintTokenEventAsset . MintTokenEvent.mintAssetAssetName
+      , -- NOTE: The query handler converts negative values to positive ones, since it considers burn events only.
+        -e ^. MintTokenEvent.mintTokenEventAsset . MintTokenEvent.mintAssetQuantity
+      , e ^. MintTokenEvent.mintTokenEventAsset . MintTokenEvent.mintAssetRedeemer
```
Comment (berewt) [inline marconi-sidechain-experimental/test/Spec/Marconi/Sidechain/Experimental/Utils.hs:206]: to `MintTokenEvent` in `test-lib`.

**Review by berewt — APPROVED** [review]
A few cosmetic and module scope changes. If you have any other ideas to remove stuff from `Utils`, they are welcome. `Utils` is usually not a great module name (even if here, I understand that it's hard to find any other host for the functions that are left in it).

**Comment by brendanrbrown** [comment]
Thanks for the review. I moved the functions that I could out of the `Utils` module and into better locations, as suggested.

Some others can go in a new `test-lib` within `marconi-chain-index`, which will also allow us to remove the `Utils` module both in that package and in `marconi-sidechain-experimental`.

I'll defer that to the ticket creating `marconi-cardano-chain-index`.

**Pull request merged**